membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -63,6 +69,42 @@ class AttributeController extends Controller with LazyLogging {
       }
     }
 
+  private def attributesFromZuora(id: String, zuoraRestService: ZuoraRestService[Future], subscriptionService: SubscriptionService[Future]): Future[Option[Attributes]] = {
+    def queryToAccountIds(response: QueryResponse): List[AccountId] =  response.records.map(_.Id)
+
+    def getSubscriptions(accountIds: List[AccountId]): Future[List[Subscription[AnyPlan]]] = {
+      def sub(accountId: AccountId): Future[List[Subscription[AnyPlan]]] = {
+        subscriptionService.subscriptionsForAccountId[AnyPlan](accountId)(anyPlanReads)
+      }
+      Future.traverse(accountIds)(id => sub(id)).map(_.flatten)
+    }
+
+    val attributes: DisjunctionT[Future, String, Option[Attributes]] = for {
+      accounts <- EitherT(zuoraRestService.getAccounts(id))
+      accountIds = queryToAccountIds(accounts)
+      subscriptions <- EitherT[Future, String, List[Subscription[AnyPlan]]](getSubscriptions(accountIds).map(a => \/.right(a)))
+    } yield {
+      AttributesMaker.attributes(id, subscriptions, LocalDate.now())
+    }
+
+    attributes.run.map(_.toOption).map(_.getOrElse(None))
+  }
+
+  private def zuoraLookup(endpointDescription: String) =
+    backendAction.async { request =>
+      authenticationService.userId(request) match {
+        case Some(id) =>
+          attributesFromZuora(id, request.touchpoint.zuoraRestService, request.touchpoint.subService).map {
+            case Some(attrs) =>
+              logger.info(s"$id is a contributor - $endpointDescription - $attrs")
```

Comment:
As the user could be a member or a contributor (or both!), it might be better to have something like: 

s"Successfully retrieved attributes from Zuora for user $id: $attrs"